### PR TITLE
Campaign added, OAuth retry

### PR DESCRIPTION
**Pull request by BrianRossmajer**
Just contributing back the changes we've needed to use node-marketo in our environment.  I hope someone finds them useful.

### CODE REVIEW

File: lib/api/campaign.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    options = _.extend({}, options, {
+      input: { leads: leads, tokens: tokens },
+      _method: 'POST'
+    });
+    options = util.formatOptions(options);
+
+    return this._connection.post(util.createPath('campaigns',campaignId,'trigger.json'), 
+	{data: JSON.stringify(options), headers: {'Content-Type': 'application/json'}});
```
Comment (peaches) [inline lib/api/campaign.js]: Please add spaces for parameters to `createPath`

File: lib/api/campaign.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  getCampaigns: function(options) {
+    var path = util.createPath( 'campaigns.json' );
+    options = _.extend({}, options, {
+      _method: 'GET'
```
Comment (peaches) [inline lib/api/campaign.js]: I don't believe `_method: 'GET'` is necessary when you're doing a GET call. This is only necessary if you're doing `this._connection.post` and want to treat it like a GET.

File: lib/api/campaign.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  request: function(campaignId, leads, tokens, options) {
+    if (!_.isArray(leads)) {
+      var msg = 'leads needs to be an Array';
+      log.error(msg);
+      return Promise.reject(msg);
+    }
+
+    options = _.extend({}, options, {
+      input: { leads: leads, tokens: tokens },
+      _method: 'POST'
```
Comment (peaches) [inline lib/api/campaign.js]: `_method` is unnecessary here.

**Comment by peaches** [comment]
This is great work, can you also add some tests?

File: lib/api/campaign.js (hunk 2 above)
Comment (BrianRossmajer) [inline lib/api/campaign.js]: I suppose I'm accustomed to being explicit about it, so anyone looking in future would know for certain what kind of call was happening.

**Comment by BrianRossmajer** [comment]
Thanks!  I can't guarantee a time-line on adding tests... I will if I have time, but didn't want to hold back on these changes in case they're useful to others.

File: lib/errors.js [hunk 4]
```diff
@@ -108,7 +108,8 @@ module.exports = {
   },
 
   isExpiredToken: function(err) {
-    return hasMarketoCode(err, mkCode.TOKEN_EXPIRED);
+    return hasMarketoCode(err, mkCode.TOKEN_EXPIRED) ||
+	   hasMarketoCode(err, mkCode.TOKEN_INVALID);
```
Comment (jdrake) [inline lib/errors.js]: :+1: I just ran into this yesterday

**Comment by peaches** [comment]
I'll merge this, I'll find some time to add the tests later.

**Pull request merged**